### PR TITLE
ROS2

### PR DESCRIPTION
**Pull request by klintan**
ROS2 version of the open-street-map library. 

Will add more comments, but I tested the visualizations and got some nice markers in RVIZ2, and the tests should pass for route_network as well. I'll take another look again this weekend but would love some feedback.

### CODE REVIEW

**Comment by klintan** [comment]
@SteveMacenski Not sure if you are a maintainer of this particular repo ? This is should be ready for review now. I'm not completely sure how to get the `colcon test` to run the `unittest` in Python. But they do run and pass running `python3 -m unittest discover tests/` 

Tried it out in Rviz2 as well and got some nice markers. But yeah let me know if you find anything (or perhaps who else we might ping to get this one going)

**Comment by SteveMacenski** [comment]
I'm not, I know absolutely nothing about this work. @3bmagyar looks like the most recent developer on it

**Comment by klintan** [comment]
> I'm not, I know absolutely nothing about this work. @bmagyar looks like the most recent developer on it

I sorry I misremembered that I saw you write something somewhere. Hopefully he can take a look. Thanks for checking in :)

**Comment by klintan** [comment]
@bmagyar just a friendly ping to see if you or another maintainer could take a look at this, would be awesome to get some maps into rviz2 for ros2.

**Comment by SteveMacenski** [comment]
Try him on discourse too.

File: route_network/setup.py [hunk 1]
```diff
@@ -1,18 +1,37 @@
...
+        ('share/' + PACKAGE_NAME, ['package.xml']),
+        (os.path.join(SHARE_DIR, "launch"), glob(os.path.join("launch", "*.launch.py"))),
+        (os.path.join(SHARE_DIR, "config"), glob(os.path.join("config", "*.yaml")))],
+    package_dir={'': 'src', },
+    py_modules=[],
+    zip_safe=True,
+    install_requires=['setuptools',
+                      'pyproj'],
+    author="Jack O'Quin",
+    maintainer="Jack O'Quin",
```
Comment (bmagyar) [inline route_network/setup.py:25]: could you add me here too please?

File: route_network/src/route_network/nodes/route_network.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        rclpy.spin(node_class)
+    except rclpy.exceptions.ROSInterruptException:
+        pass
+
+    node_class.destroy_node()
+    rclpy.shutdown()
+
+
+if __name__ == '__main__':
+    # run main function and exit
```
Comment (bmagyar) [inline route_network/src/route_network/nodes/route_network.py:162]: this is almost exactly the same as the line below, please remove

File: route_network/src/route_network/nodes/viz_routes.py [hunk 3]
```diff
@@ -55,27 +53,28 @@
...
+
+class RouteVizNode(Node):
 
     def __init__(self):
-        """ROS node to publish the route network graph for a GeographicMap.
         """
-        rospy.init_node('viz_routes')
+        ROS node to publish the route network graph for a GeographicMap.
+        """
+        super(RouteVizNode, self).__init__("viz_routes")
```
Comment (bmagyar) [inline route_network/src/route_network/nodes/viz_routes.py:63]: I think Python3 doesn't really require this anymore

https://www.journaldev.com/15911/python-super

Comment (klintan) [reply route_network/src/route_network/nodes/viz_routes.py]: Ah awesome. Fixed all the other classes as well. Thanks for taking your time to review, really appreciate it!

**Comment by klintan** [comment]
@bmagyar Updated for most of your feedback, let me know if there is anything else that needs changing.

**Pull request merged**